src/code/z_kankyo.c

```diff
@@ -897,15 +3087,246 @@ void Environment_WarpSongLeave(PlayState* play) {
     gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_SetupSkyboxStars.s")
+void Environment_SetupSkyboxStars(PlayState* play) {
+    f32 phi_f0;
+
+    if ((play->envCtx.skybox1Index == 0) && (play->skyboxId == SKYBOX_NORMAL_SKY)) {
+        if ((((void)0, gSaveContext.save.time) >= CLOCK_TIME(21, 0)) ||
+            (((void)0, gSaveContext.save.time) < CLOCK_TIME(2, 0))) {
+            phi_f0 = 1.0f;
+        } else if (((void)0, gSaveContext.save.time) > CLOCK_TIME(19, 0)) {
+            phi_f0 = 1.0f - ((CLOCK_TIME(21, 0) - ((void)0, gSaveContext.save.time)) * (1.0f / (CLOCK_TIME(2, 0) + 1)));
+        } else if (((void)0, gSaveContext.save.time) < CLOCK_TIME(3, 0)) {
+            phi_f0 = (CLOCK_TIME(3, 0) - ((void)0, gSaveContext.save.time)) * (1.0f / (CLOCK_TIME(1, 0) + 1));
+        } else {
+            phi_f0 = 0.0f;
+        }
+
+        phi_f0 = (play->envCtx.skyboxConfig == 0x18) ? 1.0f : phi_f0;
+
+        D_801F4F28 = phi_f0;
+        sEnvSkyboxNumStars = gSkyboxNumStars;
+    } else {
+        D_801F4F28 = 0.0f;
+        sEnvSkyboxNumStars = 0;
+    }
+
+    if ((sEnvSkyboxNumStars != 0) && (D_801F4F28 != 0.0f)) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        sSkyboxStarsDList = POLY_OPA_DISP;
+
+        gSPNoOp(POLY_OPA_DISP++);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    } else {
+        sSkyboxStarsDList = NULL;
+    }
+}
+
+void Environment_DrawSkyboxStar(Gfx** gfxp, f32 x, f32 y, s32 width, s32 height) {
+    Gfx* gfx = *gfxp;
+    u32 xl = x * 4.0f;
+    u32 yl = y * 4.0f;
+    u32 xd = width;
+    u32 yd = height;
+
+    gSPTextureRectangle(gfx++, xl, yl, xl + xd, yl + yd, 0, 0, 0, 0, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSkyboxStar.s")
+    *gfxp = gfx;
+}
 
+#ifdef NON_MATCHING
+void Environment_DrawSkyboxStarsImpl(PlayState* play, Gfx** gfxP) {
```

Comment:
```suggestion
#ifdef NON_MATCHING
// non_match: the second gDPSetColor macro loads D_801DD900 out of order
void Environment_DrawSkyboxStarsImpl(PlayState* play, Gfx** gfxP) {
```
Don't have to use this exact description, but I kinda want a basic description of the problem at each function, up to you if you wanted to include decomp.me link, which could get stale